src/Disqus.php

```
@@ -11,105 +11,54 @@
     {
-        $this->container = $container;
+        $this->httpClient = HttpClient::createForBaseUri(self::DISQUS_API_BASE_URI);
 
         $this->apiKey    = $apiKey;
         $this->secretKey = $secretKey;
         $this->debug     = $debug;
-
-        if ($baseUrl) {
-            $this->baseUrl = $baseUrl;
-        }
     }
 
-    /**
-     * @param string $shortname
-     * @param array  $options
-     * @param string $fetch
-     *
-     * @return string
-     */
-    public function fetch($shortname, array $options, $fetch = 'threads/listPosts')
+    public function fetch(string $shortname, array $options, string $fetch = 'threads/listPosts'): array
     {
         $this->shortname = $shortname;
 
         $options = $this->setOptions($options);
 
         $url = $this->buildUrl($options, $fetch);
-        if ($this->container->has('knp_zend_cache.manager')) {
-            $cache = $this->container->get('knp_zend_cache.manager');
-            $cache = $cache->getCache($this->container->getParameter('knp_disqus.cache.'.$shortname));
-            $key   = sha1($url);
-            if (false === ($content = $cache->load($key))) {
-                $content = json_decode($this->httpRequest($url), true);
-
-                // we cache, even if we have a bad response
-                // sometimes disqus goes down (request times out), and we don't want to keep making
-                // this hanging request each time - just don't server-side load the comments for now
-                $cache->save($content, $key);
-            }
-        } else {
-            $content = json_decode($this->httpRequest($url), true);
-        }
+
+        $content = $this->request($url);
 
         // in case we got a bad response, fake some stuff
         if (!is_array($content) || !isset($content['response'])) {
-            $content = array('response' => false);
+            $content = ['response' => false];
```

Let's throw an exception here. Maybe a `ClientException` from Symfony's HttpClient.

---

I can't use `ClientException` because we don't have the response object here and `ClientException` requires it as argument. I did `RuntimeException` here because `ClientException` extends it.